e2e-tests/performance-tests/compare/comparator.js

```diff
@@ -0,0 +1,180 @@
+			};
+		}
+
+		return {
+			block,
+			metrics: this.compareMetrics(metrics1, metrics2),
+		};
+	}
+
+	compareMetrics(metrics1, metrics2) {
+		const allMetrics = new Set([
+			...Object.keys(metrics1),
+			...Object.keys(metrics2),
+		]);
+		return Array.from(allMetrics).map(metric =>
+			this.compareMetric(metric, metrics1[metric], metrics2[metric])
+		);
+	}
+
+	compareMetric(metric, values1, values2) {
+		if (!values1 || !values2) {
+			return {
+				metric,
+				status: 'missing',
+				statusEmoji: '❓',
+				warning: `Metric ${metric} is missing in one of the files.`,
+			};
+		}
+
+		const oldStats = this.calculateStatistics(values1.times);
+		const newStats = this.calculateStatistics(values2.times);
+		const diffMean = newStats.mean - oldStats.mean;
+		const percentChangeMean = PerformanceComparator.calculatePercentChange(
+			oldStats.mean,
+			newStats.mean
+		);
+		const status = this.determineStatus(diffMean, percentChangeMean);
+
+		return {
+			metric,
+			oldStats,
+			newStats,
+			diffMean,
+			percentChangeMean,
+			...status,
+			statusEmoji: status.statusEmoji,
+		};
+	}
+
+	calculateStatistics(times) {
+		const cleanedTimes = times.filter(time => time !== 0);
+		return {
+			mean: PerformanceComparator.calculateMean(cleanedTimes),
+			median: PerformanceComparator.calculateMedian(cleanedTimes),
+			min: PerformanceComparator.calculateMin(cleanedTimes),
+			max: PerformanceComparator.calculateMax(cleanedTimes),
+			standardDeviation:
+				PerformanceComparator.calculateStandardDeviation(cleanedTimes),
+			sampleSize: cleanedTimes.length,
+		};
+	}
+
+	static calculateMean(data) {
+		return data.reduce((sum, value) => sum + value, 0) / data.length;
+	}
+
+	static calculateMedian(data) {
+		const sortedData = [...data].sort((a, b) => a - b);
+		const mid = Math.floor(sortedData.length / 2);
+		return sortedData.length % 2 === 0
+			? (sortedData[mid - 1] + sortedData[mid]) / 2
+			: sortedData[mid];
+	}
+
+	static calculateMin(data) {
+		return Math.min(...data);
+	}
+
+	static calculateMax(data) {
+		return Math.max(...data);
+	}
+
+	static calculateStandardDeviation(data) {
+		const mean = PerformanceComparator.calculateMean(data);
+		return Math.sqrt(
+			data.reduce((sum, value) => sum + Math.pow(value - mean, 2), 0) /
+				(data.length - 1)
+		);
+	}
+
+	static calculatePercentChange(oldValue, newValue) {
+		return oldValue !== 0 ? ((newValue - oldValue) / oldValue) * 100 : 0;
+	}
+
+	determineStatus(diffMean, percentChangeMean) {
+		const isAboveThreshold = Math.abs(diffMean) > this.threshold;
+		const isAbovePercentThreshold =
+			Math.abs(percentChangeMean) > this.percentThreshold;
+
+		if (isAboveThreshold) {
+			if (!isAbovePercentThreshold) {
+				return diffMean > 0
+					? { status: 'Regressed', statusEmoji: '🐢' }
+					: { status: 'Improved', statusEmoji: '🚀' };
+			} else {
+				return { status: 'Unchanged', statusEmoji: '➖' };
+			}
+		} else {
+			return { status: 'Below threshold', statusEmoji: '➖' };
+		}
+	}
```

Comment:
```suggestion
		if (isAboveThreshold && isAbovePercentThreshold) {
			if (diffMean > 0) {
				return { status: 'Regressed', statusEmoji: '🐢' };
			} else {
				return { status: 'Improved', statusEmoji: '🚀' };
			}
		} else {
			return { status: 'Below threshold', statusEmoji: '➖' };
		}
```

maybe logic should be like this?

Also, I've copied results of last github action runs and compared them with proposed logic. Here's results:

| Block | Metric | Old Mean (s) | New Mean (s) | Difference (s) | Change (%) | Status |
|-------|--------|---------------|---------------|-----------------|------------|--------|
| Simple_HOL-PRO-95 | insert | 2.12 | 2.00 | -0.11 | -5.27% | 🚀 Improved |
| Simple_ANL-PRO-09 | insert | 3.67 | 3.09 | -0.58 | -15.93% | 🚀 Improved |
| Simple_NML-PRO-18 | insert | 2.92 | 2.79 | -0.12 | -4.25% | 🚀 Improved |
| DC/CL/IB_PBGL-PRO-60 | insert | 3.91 | 3.85 | -0.06 | -1.48% | ➖ Below threshold |
| DC/CL/IB_PBGL-PRO-97 | insert | 3.51 | 3.30 | -0.21 | -6.10% | 🚀 Improved |
| DC/CL/IB_WCL-PRO-01 | insert | 3.74 | 3.53 | -0.21 | -5.48% | 🚀 Improved |
| IB_PIL-PRO-75 | insert | 2.12 | 1.93 | -0.20 | -9.24% | 🚀 Improved |
| IB_PIL-PRO-82 | insert | 3.78 | 3.51 | -0.27 | -7.07% | 🚀 Improved |
| IB_PIL-PRO-83 | insert | 3.07 | 3.10 | 0.03 | 0.97% | ➖ Below threshold |
| Complex_HOL-PRO-62 | insert | 4.16 | 4.00 | -0.16 | -3.90% | 🚀 Improved |
| Complex_SML-PRO-134 | insert | 4.93 | 4.86 | -0.07 | -1.35% | ➖ Below threshold |
| Complex_PFL-PRO-62 | insert | 3.30 | 3.29 | -0.00 | -0.04% | ➖ Below threshold |
| Pages_DSP-PRO-01 | insert | 26.97 | 25.20 | -1.77 | -6.57% | 🚀 Improved |
| Pages_GMP-PRO-01 | insert | 23.15 | 22.23 | -0.91 | -3.95% | 🚀 Improved |
| Pages_NGOP-PRO-02 | insert | 26.05 | 25.18 | -0.86 | -3.31% | 🚀 Improved |
| Button Maxi | insert | 0.45 | 0.49 | 0.03 | 7.29% | ➖ Below threshold |
| Container Maxi with Row and Columns Maxi | insert | 1.79 | 1.65 | -0.14 | -7.69% | 🚀 Improved |
| Divider Maxi | insert | 0.26 | 0.25 | -0.00 | -1.53% | ➖ Below threshold |
| Group Maxi | insert | 0.24 | 0.24 | -0.00 | -0.32% | ➖ Below threshold |
| Image Maxi | insert | 0.82 | 0.88 | 0.06 | 7.50% | ➖ Below threshold |
| Map Maxi | insert | 0.46 | 0.55 | 0.08 | 17.68% | ➖ Below threshold |
| Number Counter Maxi | insert | 0.30 | 0.32 | 0.02 | 6.45% | ➖ Below threshold |
| Icon Maxi | insert | 0.58 | 0.58 | 0.01 | 1.33% | ➖ Below threshold |
| Text Maxi | insert | 0.28 | 0.32 | 0.04 | 14.03% | ➖ Below threshold |
| Slider Maxi | insert | 1.94 | 2.03 | 0.09 | 4.60% | ➖ Below threshold |
| Accordion Maxi | insert | 1.07 | 1.09 | 0.03 | 2.56% | ➖ Below threshold |
| Video Maxi | insert | 0.49 | 0.48 | -0.00 | -0.25% | ➖ Below threshold |
| Search Maxi | insert | 0.38 | 0.41 | 0.03 | 7.07% | ➖ Below threshold |

Reply:
> maybe logic should be like this?

The logic should help us to avoid having big %% and "unchanged" at the same time, so we need all the values with Absolute difference < 0.1s to be " Below threshold". I don't think we can have "unchanged" and " Below threshold". at the same tome, so yeah, yours is better.